app/parse.py

```diff
@@ -15,8 +28,60 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description"
+        ).text.replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".ws-icon.ws-icon-star")),
+        num_of_reviews=int(
+            product_soup.select_one("p.pull-right").text.split()[0]
+        )
+    )
```

Comment:
It`s not a mistake, but you can use class By from selenium instead using beautifulsoup methods